sway-core/src/semantic_analysis/ast_node/declaration.rs

```diff
@@ -350,8 +380,12 @@ impl TypedDeclaration {
             EnumDeclaration(TypedEnumDeclaration { visibility, .. })
             | ConstantDeclaration(TypedConstantDeclaration { visibility, .. })
             | FunctionDeclaration(TypedFunctionDeclaration { visibility, .. })
-            | TraitDeclaration(TypedTraitDeclaration { visibility, .. })
-            | StructDeclaration(TypedStructDeclaration { visibility, .. }) => *visibility,
+            | TraitDeclaration(TypedTraitDeclaration { visibility, .. }) => *visibility,
+            StructDeclaration(decl_id) => {
+                de_get_struct(decl_id.clone(), &self.span())
+                    .unwrap()
```

Comment:
We should avoid `.unwrap()` whenever possible, even if it means modifying these methods to return `Result` or `CompileResult`. In my PR I solved this problem like this: https://github.com/FuelLabs/sway/blob/2f0c106b5c329ccee12058a62ab5f6b0199526c2/sway-core/src/semantic_analysis/ast_node/declaration.rs#L336